src/main/java/com/gu/logback/appender/kinesis/JsonKinesisAppender.java

```diff
@@ -0,0 +1,33 @@
+package com.gu.logback.appender.kinesis;
+
+import ch.qos.logback.core.spi.DeferredProcessingAware;
+
+import java.util.Iterator;
+import java.util.Map;
+import javax.json.*;
+
+public class JsonKinesisAppender <Event extends DeferredProcessingAware>
+        extends KinesisAppender<Event> {
+
+    public JsonKinesisAppender(Map<String, String> standardTags) {
+        super();
+        this.standardTags = standardTags;
+    }
+
+    private Map<String, String> standardTags;
+
+    @Override
+    protected void putMessage(String message) throws Exception {
+        // build a json object
+        JsonObjectBuilder json = Json.createObjectBuilder();
+        Iterator i = standardTags.keySet().iterator();
+        while (i.hasNext()) {
+            String k = (String) i.next();
+            String v = standardTags.get(k);
+            json.add(k, v);
+        }
+        json.add("message", message);
+        super.putMessage(json.build().toString());
+    }
+
+}
```

Comment:
This seems to limit you to ssa plus message as fields? We've generally used it in conjunction with `LogstashLayout` from [logstash-logback-encoder](https://github.com/logstash/logstash-logback-encoder) which lets you add extra key-value pairs at the point of logging and deals with the JSON formatting for you. Is that too heavyweight?

Reply:
It'll put whatever tags you specify in the map onto every message you send.  For us, that'll typically be S/S/A, but it could be anything.

The entire idea was to avoid changing the code at the point of logging.